### PR TITLE
Filter comments to count actual rendered elements

### LINKED ISSUE #4114: [Bug Report] list__tile__action--stack class applied with conditional rendering
### Versions and Environment
**Vuetify:** 1.1.0-alpha.6
**Vue:** 2.5.16
**Browsers:** Chrome 67.0.3396.40
**OS:** Mac OS 10.13.4

### Steps to reproduce
- Create simple v-list component with two v-btn inside v-list-tile-action
- Use v-if="true" and one button and v-if="false" on the other button

### Expected Behavior
The class list__tile__action--stack should not be applied to the single rendered button so it's displayed in the middle

### Actual Behavior
The list__tile__action--stack class is applied so the button renders on top of the list tile

### Reproduction Link
[https://codepen.io/bierik/pen/qYwdao](https://codepen.io/bierik/pen/qYwdao)

### Other comments
When using v-if="true" and v-else, it surprisingly working
There seems to be something wrong with the child count on https://github.com/vuetifyjs/vuetify/blob/v1.0.17/src/components/VList/VListTileAction.js#L8

### PR DESCRIPTION
**Pull request by ebisbe**
## Motivation and Context

Fixes #4114 

## How Has This Been Tested?

unit ( snapshot)

## Markup:

<details>

```vue
<template>
  <v-app id="inspire">
    <v-list two-line>
      <v-list-tile>
        <v-list-tile-content>
          <v-list-tile-title>some title</v-list-tile-title>
        </v-list-tile-content>
        <v-list-tile-action>
          <v-list-tile-action-text>Something</v-list-tile-action-text>
          <v-btn v-if="true">a</v-btn>
          <v-btn v-if="false">b</v-btn>
        </v-list-tile-action>
      </v-list-tile>
    </v-list>

    <v-list two-line>
      <v-list-tile>
        <v-list-tile-content>
          <v-list-tile-title>some title</v-list-tile-title>
        </v-list-tile-content>
        <v-list-tile-action>
          <v-btn v-if="true">a</v-btn>
          <v-btn v-if="false">b</v-btn>
        </v-list-tile-action>
      </v-list-tile>
    </v-list>


    <v-list two-line>
      <v-list-tile>
        <v-list-tile-content>
          <v-list-tile-title>some title</v-list-tile-title>
        </v-list-tile-content>
        <v-list-tile-action>
          <v-btn v-if="true">a</v-btn>
        </v-list-tile-action>
      </v-list-tile>
    </v-list>

    <v-list two-line>
      <v-list-tile>
        <v-list-tile-content>
          <v-list-tile-title>some title</v-list-tile-title>
        </v-list-tile-content>
        <v-list-tile-action>
          <v-list-tile-action-text>Hola</v-list-tile-action-text>
          <v-btn v-if="true">a</v-btn>
          <v-btn v-else>b</v-btn>
        </v-list-tile-action>
      </v-list-tile>
    </v-list>
  </v-app>
</template>

<script>
export default {
  data() {
    return {};
  }
};
</script>

```
</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have created a PR in the [documentation](https://github.com/vuetifyjs/vuetifyjs.com) with the necessary changes.

### CODE REVIEW

**Comment by KaelWD** [comment]
Maybe I shouldn't've marked that "good first issue", it relies on knowledge of some vue internals.

**Comment by ebisbe** [comment]
What vue internals?
Test with the playground also? What do you mean with the second phrase?

File: src/components/VList/VListTileAction.js [hunk 1]
```diff
@@ -6,7 +6,8 @@ export default {
 
   render (h, { data, children }) {
     data.staticClass = data.staticClass ? `v-list__tile__action ${data.staticClass}` : 'v-list__tile__action'
-    if ((children || []).length > 1) data.staticClass += ' v-list__tile__action--stack'
+    const filteredChild = children.filter(VNode => VNode.isComment === false)
+    if ((filteredChild || []).length > 1) data.staticClass += ' v-list__tile__action--stack'
```
Comment (johnleider) [inline src/components/VList/VListTileAction.js]: filter will always return an array, you can remove the `||`

**Comment by johnleider** [comment]
Can you add a playground?

**Comment by ebisbe** [comment]
Added Playground and filtered out empty VNodes to make the count

**Comment by ebisbe** [comment]
@johnleider Anything more to do?

**Pull request merged**